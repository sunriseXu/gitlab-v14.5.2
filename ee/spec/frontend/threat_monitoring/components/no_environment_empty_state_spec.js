import { GlEmptyState, GlSprintf } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import {
  EMPTY_STATE_DESCRIPTION,
  NO_ENVIRONMENT_TITLE,
} from 'ee/threat_monitoring/components/constants';
import NoEnvironmentEmptyState from 'ee/threat_monitoring/components/no_environment_empty_state.vue';

const documentationPath = '/docs';
const emptyStateSvgPath = '/svgs';

describe('NoEnvironmentEmptyState component', () => {
  let wrapper;

  const findGlEmptyState = () => wrapper.find(GlEmptyState);
  const findGlSprintf = () => wrapper.find(GlSprintf);

  const factory = () => {
    wrapper = shallowMount(NoEnvironmentEmptyState, {
      provide: {
        documentationPath,
        emptyStateSvgPath,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('default state', () => {
    beforeEach(() => {
      factory();
    });

    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('shows the GlEmptyState component', () => {
      expect(findGlEmptyState().exists()).toBe(true);
      expect(findGlEmptyState().attributes()).toMatchObject({
        title: NO_ENVIRONMENT_TITLE,
        svgpath: emptyStateSvgPath,
      });
    });

    it('shows the message', () => {
      expect(findGlSprintf().exists()).toBe(true);
      expect(findGlSprintf().attributes('message')).toBe(EMPTY_STATE_DESCRIPTION);
    });
  });
});
