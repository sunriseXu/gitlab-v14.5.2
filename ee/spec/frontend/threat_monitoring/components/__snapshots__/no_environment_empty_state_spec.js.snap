// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoEnvironmentEmptyState component default state matches the snapshot 1`] = `
<gl-empty-state-stub
  svgpath="/svgs"
  title="No environments detected"
>
  <gl-sprintf-stub
    message="To view this data, ensure you have configured an environment for this project and that at least one threat monitoring feature is enabled. %{linkStart}More information%{linkEnd}"
  />
</gl-empty-state-stub>
`;
